doc/defining.md

```diff
@@ -0,0 +1,272 @@
+# Defining Apps, Bundles, and Components
+
+This file describes the mechanics of implementing components and bundles.
+
+This guidelines in this file are quite prescriptive, with the intent of making all components "look the same".
+This reduces cognitive load when using components -- no need to remember one component's peculiarities.
+It also allows Agent-wide changes, where we make the same formulaic change to each component.
+If a situation arises that contradicts the guidelines, then we can update the guidelines (and change all affected components).
+In fact, many of these prescriptions can be easily verified by linters.
+
+## Defining a Component
+
+A component is defined in a dedicated package under `comp/<bundlename>/...`, where `<bundlename>` names the bundle that contains the component.
```

Comment:
Subjective but I don't see a particular need to have the components in a top-level `comp/` vs using the more-conventional `pkg/comp` prefix.

Reply:
That would end up with all agent code under `pkg/comp`.  That's fine, I guess, but not appreciably different from a top-level directory.  Absent any concrete justification, I'd like to avoid bikeshedding on this.